File: crop_image.mjs
import { Jimp } from 'jimp';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const inputFile = String.raw`C:\Users\Pietro\.gemini\antigravity\brain\28250d29-1e39-4173-888f-00055a4ff8d7\uploaded_image_1763608792454.png`;
const outputFile = path.join(__dirname, 'assets', 'book_cover_cropped.png');

async function cropImage() {
    try {
        console.log('Reading image...');
        const image = await Jimp.read(inputFile);
        
        console.log('Original size:', image.bitmap.width, image.bitmap.height);
        
        // Auto-crop
        image.autocrop({
            tolerance: 0.1,
            cropOnlyFrames: false,
            cropSymmetric: false,
            leaveBorder: 0
        });
        
        console.log('Cropped size:', image.bitmap.width, image.bitmap.height);
        
        await image.write(outputFile);
        console.log('Image saved to:', outputFile);
        
    } catch (err) {
        console.error('Error processing image:', err);
        // Fallback: try default export if named export failed (though import { Jimp } usually works for v1)
    }
}

cropImage();

File: crop_image.py
from PIL import Image, ImageChops

def auto_crop(image_path, output_path):
    try:
        img = Image.open(image_path)
        img = img.convert("RGB")
        
        # Get the background color from the top-left pixel
        bg = Image.new(img.mode, img.size, img.getpixel((0, 0)))
        
        # Calculate the difference between the image and the background
        diff = ImageChops.difference(img, bg)
        diff = ImageChops.add(diff, diff, 2.0, -100)
        
        # Get the bounding box of the non-background content
        bbox = diff.getbbox()
        
        if bbox:
            cropped_img = img.crop(bbox)
            cropped_img.save(output_path)
            print(f"Image cropped successfully to {bbox}")
        else:
            print("Could not determine bounding box, saving original.")
            img.save(output_path)
            
    except Exception as e:
        print(f"Error cropping image: {e}")

# Use the uploaded image
input_file = r"C:\Users\Pietro\.gemini\antigravity\brain\28250d29-1e39-4173-888f-00055a4ff8d7\uploaded_image_1763608792454.png"
output_file = r"c:\Users\Pietro\.gemini\antigravity\scratch\trafego_organico\assets\book_cover.png"

auto_crop(input_file, output_file)
